### ISSUE TITLE
TypedDict totatity ignored when unpacked for function call

### ISSUE THREAD

**Issue by Zeckie** [python/mypy#12754]
**Bug Report**

When unpacking a typeddict for a function call, mypy doesn't seem to check the totality of the typeddict, and instead assumes the keys are all present.

```
from typing import TypedDict

def my_function(foo: int, bar: int, baz: str) -> None:
    pass

class MyTypedDict(TypedDict, total=False):
    foo: int
    bar: int
    baz: str
    
    
a: MyTypedDict = {'bar': 1}

my_function(**a) # Should be error here, as a doesn't contain foo or baz
```

https://mypy-play.net/?mypy=latest&python=3.10&flags=strict&gist=5444e1100000a8f1678fe9cbcfc23921

**Expected Behavior**

Mypy should not assume that keys are present when the TypedDict has `total=False`, and treat it as potentially having any combination of they keys present.

**Actual Behavior**

Mypy allows this, but when executed it gives the following error:

```
Traceback (most recent call last):
  File "<path>\typeddict1.py", line 17, in <module>
    my_function(**a)  # Should be error here, as a doesn't contain foo or baz
TypeError: my_function() missing 2 required positional arguments: 'foo' and 'baz'
```

#11753

**Comment by erictraut** [comment]
We need to consider whether it's best to err on the side of safety (with the possibility of false positives) or practicality (with the possibility of false negatives). There's no right answer, but whatever decision we make should be applied consistently.

Consider how mypy handles the unpack operator for iterables.

```python
def my_function(a: int, b, int):
    ...

def func(b: list[int]):
    my_function(*b)

func([1])
```

This type checks without errors even though it generates an exception at runtime. So mypy chose the "practical" choice here rather than the "safe" choice. For consistency, it probably makes sense that it would do the same for unpacking of TypedDicts with fields that are not required.
